var/spack/repos/builtin/packages/py-cylc-flow/package.py

```diff
@@ -29,15 +32,56 @@ class PyCylcFlow(PythonPackage):
     depends_on("py-jinja2@3.0", type=("build", "run"))
     depends_on("py-metomi-isodatetime@3.0", type=("build", "run"), when="@:8.2.0")
     depends_on("py-metomi-isodatetime@3:3.1", type=("build", "run"), when="@8.2.3:")
-    depends_on("py-protobuf@4.21.2:4.21", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"), when="@8.3:")
+    depends_on("py-protobuf@4.21.2:4.21", type=("build", "run"), when="@:8.2")
+    depends_on("py-protobuf@4.24.4:4.24", type=("build", "run"), when="@8.3:")
     depends_on("py-psutil@5.6.0:", type=("build", "run"))
     depends_on("py-pyzmq@22:", type=("build", "run"), when="@8.2:")
     depends_on("py-pyzmq@22", type=("build", "run"), when="@:8.1")
-    depends_on("py-importlib-metadata", type=("build", "run"), when="^python@:3.7")
-    depends_on("py-urwid@2", type=("build", "run"))
+    depends_on("py-importlib-metadata", type=("build", "run"), when="@:8.2 ^python@:3.7")
+    depends_on("py-importlib-metadata@5:", type=("build", "run"), when="@8.3: ^python@:3.11")
+    depends_on("py-urwid@2:2.6.1,2.6.4:2", type=("build", "run"))
     depends_on("py-rx", type=("build", "run"))
     depends_on("py-promise", type=("build", "run"))
     depends_on("py-tomli@2:", type=("build", "run"), when="^python@:3.10")
 
-    # Non-Python dependencies
-    depends_on("graphviz", type="run")
+    # Non-Python dependencies for creating graphs.
+    # We want at least the pangocairo variant for
+    # graphviz so that we can create output as png.
+    depends_on("graphviz+pangocairo", type="run")
+
+    # DH* KEEP ALL THIS UNTIL https://github.com/spack/spack/pull/48052 is merged.
```

Comment:
Lines 53 to 87 won't be submitted upstream. I am keeping them around for now until we know which solution the spack developers and package maintainers want.